chapters/07_chapter_6.tex

```diff
@@ -0,0 +1,390 @@
+$$[...] \rightarrow \texttt{rstsc} \rightarrow \mbox{function()} \rightarrow \texttt{rstsc} \rightarrow [...]$$
+\noindent By using this instruction, due to out-of-execution, the actual execution order could be different, resulting with wrong measurements. The solution is to use the pseudo-serializing instruction \texttt{rdtscp} or to insert a serializing instruction like \texttt{cpuid} or use to fence instructions like \texttt{mfence} \cite{benchmark2010}.
+
+We will now introduce two cache attack (main) techniques: Flush+Reload \cite{Gullasch:2011:CGB:2006077.2006784, Osvik:2006:CAC:2117739.2117741, Yarom2014} and Prime+Probe \cite{Percival2009,Osvik:2006:CAC:2117739.2117741,Liu:2015:LCS:2867539.2867673}. Both of them are exploitable on x86 and ARM and can be used for both covert channels and side-channel attacks.
+\subsection{Flush+Reload}
+\label{sec:flushreload}
+The attack is made of four basic steps and it goes as following:
+\begin{enumerate}
+    \item \textbf{Map. } The attacker maps a shared library (Illustration in Fig. \ref{fig:fr_sharedlib}), by doing that he will have a shared cache line with the victim.
+    \item \textbf{Flush.} The attacker \textit{flushes} the shared cache line, it can be done via unprivileged instruction like \texttt{clflush}.
+    \item \textbf{Victim.} The attacker lets the victim load (or not load) the shared line, depending on the victim's behaviour.
+    \item \textbf{Reload.} The attacker reloads the shared cache line. If the cache line has a \textit{hit} the attacker infers that the victim loaded the data on the previous step, if the cache line has a \textit{miss} the attacker infers that the victim did not load the data on the previous step.
+\end{enumerate}
+
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=\textwidth]{images/fr_sharedlib.JPG}
+    \caption{Attacker maps shared library (shared memory, in cache), shared cache line marked in red}
+    \label{fig:fr_sharedlib}
+\end{figure}
+
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=\textwidth]{images/fr_flow.png}
+    \caption{Flush+Reload attack flow. Between the first flush and the first reload the victim did not access the shared cache line, so the first reload resulted with a cache hit. Then, after the second flush the victim accessed the shared cache line and hence, the second reload resulted with a miss.}
+    \label{fig:fr_flow}
+\end{figure}
+
+\noindent The first step (Mapping a shared library) can be done once, while steps 2-4 are repeatable as much as the attacker wants. The main advantage of this attack technique is the fine granularity, which is 1 memory line (usually 64B). On the other hand, this technique is somewhat restrictive, it needs \texttt{clflush} instruction, which is not always available e.g., on ARM-v7 and it needs a shared memory. Illustration of the attack flow in Fig. \ref{fig:fr_flow}.
```

Comment:
Please use cleveref package and then use \cref or \Cref to get "Fig." automatically added.

Reply:
Done